implementation/jobberknoll/api/ext/routes/get-account-by-id-route.ts

```diff
@@ -0,0 +1,50 @@
+import { createRoute, type RouteHandler } from "@hono/zod-openapi";
+import type { GetAccountByIdUseCase } from "@jobberknoll/app";
+import { isOk } from "@jobberknoll/core/shared";
+import { authorize } from "~/ext/authorization.ts";
+import { UserUnauthorizedDto } from "~/ext/contracts/mod.ts";
+import { extHeadersSchema } from "~/ext/openapi.ts";
+import {
+  AccountDto,
+  AccountNotFoundDto,
+  SchemaMismatchDto,
+} from "~/shared/contracts/mod.ts";
+import { mapAccountToDto } from "~/shared/mappers/mod.ts";
+import { IdParamSchema, jsonRes } from "~/shared/openapi.ts";
+
+export const getAccountByIdRoute = createRoute({
+  method: "get",
+  path: "/accounts/{id}",
+  summary: "Get account by ID",
+  tags: ["Accounts"],
+  description: "Only for admin users.",
+  request: {
+    headers: extHeadersSchema("admin"),
+    params: IdParamSchema,
+  },
+  responses: {
+    200: jsonRes(AccountDto, "Retrieved account by ID."),
+    401: jsonRes(
+      UserUnauthorizedDto,
+      "The user does not have access to the resource.",
+    ),
+    404: jsonRes(AccountNotFoundDto, "The account could not be found."),
+    422: jsonRes(
+      SchemaMismatchDto,
+      "The request data did not align with the schema.",
+    ),
+  },
+});
+
+export function getAccountByIdHandler(
+  getAccountById: GetAccountByIdUseCase,
+): RouteHandler<typeof getAccountByIdRoute> {
+  return authorize("admin", async (c) => {
```

Comment:
**thought**: How hard would it be to make `authorize` accept multiple roles? I've checked with E2 and acording to the document you dont have to implement that.

Reply:
```diff
  export function authorize<R extends RouteConfig>(
-   expectedRole: UserRole,
+   expectedRoles: UserRole[],
    handler: RouteHandler<R>,
  ): RouteHandler<R> {
    return (c, next) => {
      const userRole = c.req.header("jp-user-role");
-     if (expectedRole !== userRole) {
+     if (!expectedRoles.includes(userRole as UserRole)) {
```

Although I won't do it, since I wouldn't ever use it.